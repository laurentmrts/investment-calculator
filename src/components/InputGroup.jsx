import SingleInput from "./SingleInput.jsx";

export default function InputGroup({ onChange, inputsToCalculate }) {
  return (
    <section id="user-input">
      <div className="input-group">
        <SingleInput inputsToCalculate={inputsToCalculate} onChange={onChange} label='initialInvestment'/>
        <SingleInput inputsToCalculate={inputsToCalculate} onChange={onChange} label='annualInvestment'/>
        <SingleInput inputsToCalculate={inputsToCalculate} onChange={onChange} label='expectedReturn'/>
        <SingleInput inputsToCalculate={inputsToCalculate} onChange={onChange} label='duration'/>
      </div>
    </section>
  );
}