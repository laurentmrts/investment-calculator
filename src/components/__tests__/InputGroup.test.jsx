import {describe, it, beforeEach, expect, vi} from "vitest";
import {render, screen} from "@testing-library/react";
import {userEvent} from "@testing-library/user-event";
import InputGroup from "../InputGroup.jsx";

describe('InputGroup component', () => {

    let onChange;

    beforeEach(() => {
        onChange = vi.fn()
        render(<InputGroup inputsToCalculate={{
            initialInvestment: 10_000,
            annualInvestment: 1200,
            expectedReturn: 5,
            duration: 10
        }} onChange={onChange} testId="testId"/>)
    })

    it('should match snapshot', async () => {
        expect(await screen.findByTestId('testId')).toMatchSnapshot()
    })

    describe.each([
        ['initialInvestment', 'INITIAL INVESTMENT', 10_000],
        ['annualInvestment', 'ANNUAL INVESTMENT', 1200],
        ['expectedReturn', 'EXPECTED RETURN', 5],
        ['duration', 'DURATION', 10]
    ])('has a SingleInput component %s', (testId, label, value) => {

        it(`has a label ${label}`, async () => {
            expect(await screen.findByTestId(`testId-${testId}-label`)).toHaveTextContent(label)
        })

        it(`has an input with initial value ${value}`, async () => {
            const input = await screen.findByTestId(`testId-${testId}-input`)
            expect(input).toHaveValue(value)
            await userEvent.clear(input)
            expect(onChange).lastCalledWith(testId, '')
        })

    })

})
