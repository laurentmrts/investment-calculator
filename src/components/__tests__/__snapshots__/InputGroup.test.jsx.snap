// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputGroup component > should match snapshot 1`] = `
<div
  class="input-group"
  data-testid="testId"
>
  <p
    data-testid="testId-initialInvestment"
  >
    <label
      data-testid="testId-initialInvestment-label"
    >
      INITIAL INVESTMENT
    </label>
    <input
      data-testid="testId-initialInvestment-input"
      required=""
      type="number"
      value="10000"
    />
  </p>
  <p
    data-testid="testId-annualInvestment"
  >
    <label
      data-testid="testId-annualInvestment-label"
    >
      ANNUAL INVESTMENT
    </label>
    <input
      data-testid="testId-annualInvestment-input"
      required=""
      type="number"
      value="1200"
    />
  </p>
  <p
    data-testid="testId-expectedReturn"
  >
    <label
      data-testid="testId-expectedReturn-label"
    >
      EXPECTED RETURN
    </label>
    <input
      data-testid="testId-expectedReturn-input"
      required=""
      type="number"
      value="5"
    />
  </p>
  <p
    data-testid="testId-duration"
  >
    <label
      data-testid="testId-duration-label"
    >
      DURATION
    </label>
    <input
      data-testid="testId-duration-input"
      required=""
      type="number"
      value="10"
    />
  </p>
</div>
`;
